BlueprintUI/Tests/FlowTests.swift

```diff
@@ -0,0 +1,77 @@
+import XCTest
+@testable import BlueprintUI
+
+
+class FlowTests: XCTestCase {
+
+    func test_measurement() {
+
+        ///
+        /// Zero size tests
+        ///
+
+        XCTAssertEqual(Flow {}.content.measure(in: .unconstrained), .zero)
+        XCTAssertEqual(Flow {}.content.measure(in: .init(CGSize(width: 10, height: 10))), .zero)
+
+        ///
+        /// One element tests
+        ///
+
+        let one = Flow {
+            ConstrainedSize(width: 50, height: 5)
+        }
+
+        XCTAssertEqual(
+            one.content.measure(in: .unconstrained),
+            CGSize(width: 50, height: 5)
+        )
+
+        /// Note: We don't manually constrain measured elements to the size of
+        /// the input constraint, similar to stacks and other elements.
+
+        XCTAssertEqual(
+            one.content.measure(in: .init(CGSize(width: 40, height: 4))),
+            CGSize(width: 50, height: 5)
+        )
+
+        ///
+        /// Multiple element tests
+        ///
+
+        let multiple = Flow {
+            ConstrainedSize(width: 50, height: 5)
+            ConstrainedSize(width: 100, height: 5).flowChild(key: "aKey")
+            ConstrainedSize(width: 50, height: 5)
+        }
+
+        XCTAssertEqual(
+            multiple.content.measure(in: .init(width: 110)),
+            CGSize(width: 100, height: 15)
+        )
+
+        ///
+        /// Multiple element tests, but the elements are too wide.
+        /// Ensure that they don't end up creating extra rows.
+        ///
+
+        let multipleTooWide = Flow(rowSpacing: 5) {
+            ConstrainedSize(width: 110, height: 5)
+            ConstrainedSize(width: 120, height: 5).flowChild(key: "aKey")
+            ConstrainedSize(width: 130, height: 5)
+        }
+
+        XCTAssertEqual(
+            multipleTooWide.content.measure(in: .init(width: 100)),
+            CGSize(width: 130, height: 25)
```

Comment:
I guess this unit test is related to my comment [here](https://github.com/square/Blueprint/pull/489/files#r1578492578). I'm not sure I fully understand this. It seems like it would cause the container to fit the widest element? If so, wouldn't it violate the size constraint that it's being provided? I'm thinking of scenarios where a pill containing text can grow larger than a developer might expect because of localisation and/or dynamic type.

Reply:
I flip-flopped on this one – this mirrors behavior of `Column`, which doesn't constrain the measurement, _except_ in `.fill` alignment mode. Eg, this passes:

```swift
XCTAssertEqual(
    Column {
        ConstrainedSize(width: 110, height: 5)
    }.content.measure(in: .init(width: 100)),
    CGSize(width: 110, height: 5)
)
```

But this does not (the measurement in this case is 100):

```swift
XCTAssertEqual(
    Column(alignment: .fill) {
        ConstrainedSize(width: 110, height: 5)
    }.content.measure(in: .init(width: 100)),
    CGSize(width: 110, height: 5)
)
```

I'm good to convert this to constrain the contained children, since yeah, it is a bit odd for things to overflow the container and basically never what someone wants.

- [ ] Convert

Reply:
Even in the Column example, is the expansion the desired behaviour? If a label is longer than anticipated in the non-fill case, would it truncate, go into the next line (assuming `numLines` > 1), or cause the column to expand?

Reply:
This is done